desc/optimize/tr_subproblems.py

```diff
@@ -421,26 +421,20 @@ def truefun(*_):
     def falsefun(*_):
         alpha_upper = jnp.linalg.norm(J.T @ f) / trust_radius
         alpha_lower = 0.0
-        alpha = setdefault(
-            initial_alpha,
-            jnp.maximum(0.001 * alpha_upper, (alpha_lower * alpha_upper) ** 0.5),
-        )
+        alpha = setdefault(initial_alpha, alpha_lower)
         k = 0
         # algorithm 4.3 from Nocedal & Wright
         fp = jnp.pad(f, (0, J.shape[1]))
 
         def loop_cond(state):
-            alpha, alpha_lower, alpha_upper, phi, k = state
+            p, alpha, alpha_lower, alpha_upper, phi, k = state
             return (jnp.abs(phi) > rtol * trust_radius) & (k < max_iter)
 
         def loop_body(state):
-            alpha, alpha_lower, alpha_upper, phi, k = state
+            p, alpha, alpha_lower, alpha_upper, phi, k = state
 
-            alpha = jnp.where(
-                (alpha < alpha_lower) | (alpha > alpha_upper),
-                jnp.maximum(0.001 * alpha_upper, (alpha_lower * alpha_upper) ** 0.5),
-                alpha,
-            )
+            alpha = jnp.where((alpha < alpha_lower), alpha_lower, alpha)
+            alpha = jnp.where((alpha > alpha_upper), alpha_upper, alpha)
```

Comment:
do these changes help reduce the number of iterations of the subproblem? If so by how much? 

We may want to make the same changes to the other trust region subproblems (svd, cho etc) to keep them consistent.

Reply:
also this could be simplified to `alpha = jnp.clip(alpha, alpha_lower, alpha_upper)`

Reply:
I don't know the exact effect of clipping alpha rather than setting it to `0.001*alpha_upper`. My reasoning for these changes was to make the initial guess of alpha 0, and whenever alpha goes out of the range setting it to `0.001*alpha_upper` seemed to be taking alpha away from the actual solution. So, it seemed necessary to clip alpha when I changed the initial guess.

Reply:
The total number iterations for root finding depends on the case but these changes help to reduce it. Previously a root finding was taking 5-10 iterations, now around 3-6.

Reply:
ok in that case can you make the same changes to `trust_region_step_exact_{svd,cho}`